compilers/concrete-optimizer/concrete-optimizer/src/optimization/config.rs

```diff
@@ -76,9 +93,9 @@ impl SearchSpace {
             levelled_only_lwe_dimensions,
         }
     }
-    pub fn default(processing_unit: config::ProcessingUnit) -> Self {
+    pub fn default(processing_unit: config::ProcessingUnit, parameter_restrictions: ParameterRestrictions) -> Self {
         match processing_unit {
-            config::ProcessingUnit::Cpu => Self::default_cpu(),
+            config::ProcessingUnit::Cpu => Self::default_cpu(parameter_restrictions),
             config::ProcessingUnit::Gpu {
```

Comment:
If we want to implement that as a parameter restriction should also be ported on gpu parameter selection, I think the easy way would be to implement a `fn apply_restriction(Self, ParameterRestrictions) -> Self` and apply that to the whole expression.